src/routes/auth/facebook.js

```diff
@@ -1,11 +1,17 @@
 import express from 'express';
 import passport from '../../middlewares/passport';
 import AuthPassportController from '../../controllers/AuthPassportController';
+import checkSingUpPermission from '../../middlewares/checkSingUpPermission';
 
 const router = express.Router();
 
 router.get('/', passport.authenticate('facebook', { scope: ['email'] }));
 
-router.get('/callback', passport.authenticate('facebook'), AuthPassportController.loginOrSignup);
+router.get(
+  '/callback',
+  passport.authenticate('facebook'),
+  checkSingUpPermission,
+  AuthPassportController.loginOrSignup
```

Comment:
do we not need the async handler here?

Reply:
We don't need the async handler here, this feature is for Jonathan and he handles errors in a helper function.